auth.php

```diff
@@ -172,6 +172,15 @@ public function user_login_userkey() {
         $this->userkeymanager->delete_keys($key->userid);
 
         $user = get_complete_user_data('id', $key->userid);
+        if (!empty($user->suspended)) {
+            $failurereason = AUTH_LOGIN_SUSPENDED;
+
+            // Trigger login failed event.
+            $event = \core\event\user_login_failed::create(array('userid' => $user->id,
+                'other' => array('username' => $user->username, 'reason' => $failurereason)));
+            $event->trigger();
+            throw new invalid_parameter_exception('User suspended');
```

Comment:
If this error will be displayed to a user, can we please replace this moodle_exception + add a lang string so it could be localised.